controller/src/beerocks/master/db/node.h

```diff
@@ -183,6 +187,30 @@ class node {
     };
     std::shared_ptr<radio> hostap;
 
+    class link_metrics_data {
+    public:
+        link_metrics_data(){};
+        ~link_metrics_data(){};
+
+        std::vector<ieee1905_1::tlvTransmitterLinkMetric::sInterfacePairInfo>
+            transmitterLinkMetrics;
+        std::vector<ieee1905_1::tlvReceiverLinkMetric::sInterfacePairInfo> receiverLinkMetrics;
+
+        bool add_transmitter_link_metric(
+            std::shared_ptr<ieee1905_1::tlvTransmitterLinkMetric> TxLinkMetricData);
+        bool add_receiver_link_metric(
+            std::shared_ptr<ieee1905_1::tlvReceiverLinkMetric> RxLinkMetricData);
+    };
+
+    /*
+    * This map holds link metric "data struct" per reporting Agent sMacAddr .
+    * "data struct" holds map of the actual link_metrics_data vector (tx/rx) per reported Agent sMacAddr.
+    * Map is Used in TYPE_GW/TYPE_IRE nodes.
```

Comment:
correct IRE is not relevant here, I'll remove it from comment .
I use method  in TYPE_GW anyway.
regarding saving it on the node, it was @morantr   suggestion.

Reply:
Currently, all measurements are being saved on the node that measured them.
I would keep it that way.

Reply:
> Currently, all measurements are being saved on the node that measured them.

In this PR, that statement is not true. All measurements are saved on the gateway node, not the node that made the measurements.

It would indeed be a good idea to store the measurements in the IRE (or GW) node that made the measurements. But then, we should have just a `std::unordered_map<sMacAddr, son::node::link_metrics_data>` in the node.

It would also mean that to construct the combined infrastructure metrics, we'd have to iterate over all nodes instead of just querying this one map.

Reply:
there is no point in saving metrics on the IRE at least not for certification.
I think what @morantr meant to say is that if  GW (controller) will be using  metric data measurements then  it should be save on its node's db.